server/routes/index.js

```diff
@@ -1,8 +1,120 @@
 const express = require("express");
 const router = express.Router();
+const bcrypt = require("bcrypt");
+const jwt = require("jsonwebtoken");
+const auth = require("../middleware/auth");
+const User = require("../models/user");
 
-router.get("/welcome", function (req, res, next) {
+// testing root route
+router.get("/", function (req, res, next) {
   res.status(200).send({ welcomeMessage: "Step 1 (completed)" });
 });
 
+// register route
+router.post("/register", async (req, res) => {
+  const { email, confirmEmail, password, confirmPassword } = req.body;
+
+  try {
+    // validate if all fields are filled
+    if (!email || !confirmEmail || !password || !confirmPassword) {
+      return res.status(400).send("All fields need to be filled!");
+    }
+    // validate if email aleady exist
+    const userExist = await User.findOne({ email }).exec();
+    if (userExist) {
+      return res.status(409).send("Email already exist!");
+    }
+    // validate if password length is lesser than 6
+    if (password.length < 6) {
+      return res
+        .status(401)
+        .send("Password needs to be at least 6 characters long");
+    }
+    // validate confirm email and confirm password
+    if (email !== confirmEmail || password !== confirmPassword) {
+      return res
+        .status(401)
+        .send(
+          "Email and confirm Email or Password and confirm Password do not match"
+        );
+    }
+    // hash password - bcrypt
+    const salt = await bcrypt.genSalt();
+    const hashedPassword = await bcrypt.hash(password, salt);
+    // create new user
+    const newUser = new User({
+      email,
+      password: hashedPassword,
+    });
+    await newUser.save();
+    // send token
+    // proess.env.JWT_KEY to store secret key
+    const secretKey = process.env.JWT_KEY || "secret";
+    const token = jwt.sign(
+      {
+        id: newUser._id,
+        email: newUser.email,
+      },
+      secretKey,
+      {
+        expiresIn: "1hr",
+      }
+    );
+    // save token in cookie
+    // process.env.COOKIE_NAME to store cookie name
+    const cookieName = process.env.COOKIE_NAME || "token";
+    // can add maxAge for cookie expiration time
+    return res.cookie(cookieName, token, { httpOnly: true }).status(201).send({
```

Comment:
Your cookies need an expiry as well

Also, we may also want to add the options secure = true for the PROD app only (can check the environment)